core/shapes.cpp

```diff
@@ -428,12 +435,43 @@ void BoardShape::setupInserts() {
 
 const GCodeWriter BoardShape::generateBaseCutout(const Machine& machine) {
   auto tool = machine.tool(machine.baseCutoutTool());
-  auto paths = PathUtils::OffsetPath(buildOverallPath(machine), -0.2);
+  
+  // 1. Get path assuming full wrap of metal edge
+  auto fullWrapPath = PathUtils::OffsetPath(buildOverallPath(machine), -0.2);
+  assert(fullWrapPath.size() == 1);
+  
+  // 2. Form square outlining limit of metal edges (whole board if no partial edges)
+  MCFixed noseEdgeX = 0;
+  if (m_noseEdgeExt != 999.0) {
+    noseEdgeX += (m_noseLength - m_noseEdgeExt);
+  }
+  MCFixed tailEdgeX = m_noseLength + m_effectiveEdge + m_tailLength;
+  if (m_tailEdgeExt != 999.0) {
+    tailEdgeX -= (m_tailLength - m_tailEdgeExt);
+  }
+  Path edgeLimitSquare;
+  edgeLimitSquare.push_back(Point(noseEdgeX, -(m_noseWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(tailEdgeX, -(m_tailWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(tailEdgeX, (m_tailWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(noseEdgeX, (m_noseWidth/2 + 1)));
+  edgeLimitSquare.push_back(Point(noseEdgeX, -(m_noseWidth/2 + 1)));
+  
```

Comment:
Interesting... test this with a board with a lot of taper, like multiple cm. I think you'll find that you're clipping off more of the base than you intend at the hips. If the nose and tail are the same width, this makes a square. Otherwise, it makes a trapezoid. And that might not contain all of the path you intend. You probably want to take the max of the nose and tail widths, and use that to form your square.